### ISSUE TITLE
Is there a way to prevent the local refresh token from overwriting the new refresh token when refreshing session?

### ISSUE THREAD

**Issue by Rob117** [aws-amplify/amplify-js#4467]
***Problem:***

I have a local refresh token. When I call [refreshSession()](https://github.com/aws-amplify/amplify-js/blob/86597db2b0704d7aff5b612557536142b82e1731/packages/amazon-cognito-identity-js/src/CognitoUser.js#L1377), the API actually returns a new refresh token. However, it appears that the incoming, newer refresh token [is being overwritten by the current, local token](https://github.com/aws-amplify/amplify-js/blob/86597db2b0704d7aff5b612557536142b82e1731/packages/amazon-cognito-identity-js/src/CognitoUser.js#L1414).

***Solution:***

Is there an option to simply use the incoming refresh token - which I imagine should be newer and thus last longer? Or, alternatively, is there a reason why the incoming token is being overwritten by the local copy?

**Comment by haverchuck** [comment]
@Rob117 I don't believe there is a way to accomplish this OOTB, so I'll leave this marked as a feature request.

**Comment by Rob117** [comment]
@haverchuck 
Thank you, I appreciate it.

**Comment by jcary741** [comment]
I believe this is a duplicate of [#10319](https://github.com/aws-amplify/amplify-js/issues/10319), the conclusion of which was "That's not how OAuth works"
